autodoc.pl

```diff
@@ -404,7 +404,9 @@ ($;$)
 }
 
 sub check_and_add_proto_defn {
-    my ($element, $file, $line_num, $raw_flags, $ret_type, $args_ref) = @_;
+    my ($element, $file, $line_num, $raw_flags, $ret_type, $args_ref,
+        $definition_type
+       ) = @_;
```

Comment:
perhaps it's time for function signatures (not requiring any changes here)

Reply:
I am always unsure what is going to work with miniperl and what won't; so to save time I don't try anything.

Reply:
`use v5.40.0` will enable function signatures and other features, doesn't load any modules and is safe for `miniperl`.  `use feature` does load `feature.pm` but doesn't need XS and is safe for `miniperl`.

I generally prefer the `use v...` syntax since it's a little more efficient and you get the exact feature set (including disabling switch, indirect etc) as opposed to the "only switch features on" behaviour of `use feature`.

Just for future consideration.